Adapt Activity & Privacy Stats widgets for ads and trackers

noisysocks opened the pull request:
**Asana Task/Github Issue:** https://app.asana.com/0/1209121419454298/1209865676528005/f

## Description

| `adBlocking.enabled = false` | `adBlocking.enabled = true` |
| - | - | 
|  ![none-noAdBlocking](https://github.com/user-attachments/assets/72a1578e-37d5-40d0-853e-55d37a9ddb4f) | ![none-adBlocking](https://github.com/user-attachments/assets/4004fdda-f3f1-4282-b8a8-a9292a6bab87) |
| ![some-noAdBlocking](https://github.com/user-attachments/assets/0a0f8982-2ba3-4f78-afce-53b8866e86c0) | ![some-adBlocking](https://github.com/user-attachments/assets/fffe5a2d-6f12-4bd7-8f53-ee341bc8c2b6) |
| ![many-noAdBlocking](https://github.com/user-attachments/assets/d39ff8a3-f97a-437a-92f4-8e50eb8b28d6) | ![many-adBlocking](https://github.com/user-attachments/assets/cda2d5d6-bf18-4806-bd22-45fadf3fdfc0) |

Updates the Activity and Privacy Stats widgets to support a variation where browser protections include ad blocking as well as tracker protection.

- Updates UI text, icons, and styles based on whether the browser blocks just trackers or ads and trackers.
- Introduces a `adBlocking` feature flag to `NewTabPageSettings` to control display.
- Updates tests and mock data accordingly.

## Testing Steps

Navigate to:

- https://deploy-preview-1621--content-scope-scripts.netlify.app/build/pages/new-tab/?platform=windows&feed=both&activity=empty&stats=none&adBlocking=enabled
- https://deploy-preview-1621--content-scope-scripts.netlify.app/build/pages/new-tab/?platform=windows&feed=both&activity=onlyTopLevel&stats=some&adBlocking=enabled
- https://deploy-preview-1621--content-scope-scripts.netlify.app/build/pages/new-tab/?platform=windows&feed=both&activity=some&stats=many&adBlocking=enabled

## Checklist

<!--
  These questions are a friendly reminder to shipping code, if you're uncertain ask the AoR owners.
  It's also totally appropriate to not check some of these boxes, if they don't apply to your change.
-->
*Please tick all that apply:*

- [x] I have tested this change locally
- [x] I have tested this change locally in all supported browsers
- [x] This change will be visible to users
- [x] I have added automated tests that cover this change
- [x] I have ensured the change is gated by config
- [ ] This change was covered by a ship review
- [ ] This change was covered by a tech design
- [ ] Any dependent config has been merged

mgurgel reviewed (changes requested):
Looks excellent to me. The useTrackerType hook makes it easier to roll back the changes if the experiment is not successful. In fact, I wonder if it's worth creating separate components (e.g. ActivityHeadingAds and TrackerStatusAds) to make it easier to commit to one or the other once the experiment is done. We've been doing something similar in Duck Player to good results. Not a request, just a thought.

I agree with the choice of storing trackerType in activity config, but I'd still like to get Shane's sign-off on it when he's back.

noisysocks commented:
> I agree with the choice of storing trackerType in activity config, but I'd still like to get Shane's sign-off on it when he's back.

I may end up using a `initResponse` feature flag after all depending on outcome of https://app.asana.com/1/137249556945/project/0/task/1209958041601248.

noisysocks commented:
I’ve made some significant changes here.

The PR now updates both the Activity and Privacy Stats widgets to support the variation where the browser blocks ads as well as trackers. (Before, only the Activity widget was updated.) This addresses an oversight in the original design, see https://app.asana.com/1/137249556945/project/0/task/1209958041601248.

Because more than one type of widget is now affected by this feature flag I think it makes sense to do the flagging in `NewTabPageSettings` instead of the per-widget config object.

I’ve made those changes, updated tests, and updated the PR description.

mgurgel commented on special-pages/pages/new-tab/readme.md:
```diff
@@ -0,0 +1,28 @@
+- `customizerDrawer` - If set to `enabled`, enables the Customizer feature flag in the initial setup.
+- `autoOpen` - If `customizerDrawer` is `enabled` and this parameter is `true`, sets the `autoOpen` flag for the customizer drawer in the initial setup.
+- `defaultStyles` - If set to `visual-refresh`, applies specific background color variables (`--default-light-background-color`, `--default-dark-background-color`) to the body.
+- `adBlocking` - If set to `enabled`, configures the Activity and Privacy Stats widgets to indicate that both ads and trackers are blocked.
```
Nice 😃

mgurgel reviewed (approved):
Looks good 👍 Over to @shakyShane for an architecture review

shakyShane commented on special-pages/pages/new-tab/readme.md:
```diff
@@ -0,0 +1,28 @@
+
+### Test URL params
+
+URL parameters can be used to override the default values of the onboarding flow's configuration. Supported parameters are:
```
some 'onboarding' leftovers :)

noisysocks replied:
Oops thanks!

shakyShane reviewed (commented):
Very nice work - my only points are the same as the other PR:

- 1: what's the plan with translations?
- 2: the PR should go green once my other fix is merged.
